src/cls/lock/cls_lock_client.h

```diff
@@ -66,6 +45,28 @@ namespace rados {
                              const std::string& cookie, const std::string& tag,
                              const std::string& new_cookie);
 
+      extern int lock(librados::IoCtx *ioctx,
+		      const std::string& oid,
+		      const std::string& name, ClsLockType type,
+		      const std::string& cookie, const std::string& tag,
+		      const std::string& description, const utime_t& duration,
+		      uint8_t flags);
+
+      extern int unlock(librados::IoCtx *ioctx, const std::string& oid,
+			const std::string& name, const std::string& cookie);
+
+      extern int break_lock(librados::IoCtx *ioctx, const std::string& oid,
+			    const std::string& name, const std::string& cookie,
+			    const entity_name_t& locker);
+
+      extern int list_locks(librados::IoCtx *ioctx, const std::string& oid,
+			    list<std::string> *locks);
+
+      extern int get_lock_info(librados::IoCtx *ioctx, const std::string& oid,
+			       const std::string& name,
+			       map<locker_id_t, locker_info_t> *lockers,
+			       ClsLockType *type, std::string *tag);
+
```

Comment:
you grouped these calls together - was the intent to surround them in the #ifndef?

Reply:
yes, but I'm leaving those calls out for now because some of those functions are called by the RGW, and I want to refactor them in another PR.